source/common/upstream/upstream_impl.h

```diff
@@ -605,6 +609,8 @@ class ClusterInfoImpl : public ClusterInfo, protected Logger::Loggable<Logger::I
   mutable ClusterStats stats_;
   Stats::IsolatedStoreImpl load_report_stats_store_;
   mutable ClusterLoadReportStats load_report_stats_;
+  const std::unique_ptr<Stats::IsolatedStoreImpl> timeout_budget_stats_store_;
+  mutable absl::optional<ClusterTimeoutBudgetStats> timeout_budget_stats_;
```

Comment:
Why does this need to be mutable?

Reply:
Was cargo-culting against upstream.h to only define const pure virtual functions.  Will adjust this and the rest of the signatures.

Reply:
Immediate results are mixed -- `const ClusterInfo` is passed around, going to play with it a bit.